### DISCUSSION TITLE
PlatformIO no member named 'getTouch'

### DISCUSSION THREAD

**Discussion by MHz000** [Bodmer/TFT_eSPI#2881] [Compatible displays and setup files]
I'm using PlatfornIO and I get the error 'class TFT_ESPI' has no member named 'getTouch'. In an answer posted by @Bodmer in https://github.com/Bodmer/TFT_eSPI/issues/117#issuecomment-379113278_ Bodmer says "If you are using PlatformIO then see the note in the Tools folder of the library" . Unfortunately, there is no corresponding information to be found in the tools folder. Can anybody help me further? My platformio.ini file looks like this
<code>
build_flags =
-D USER_SETUP_LOADED
-D ILI9341_2_DRIVER
-D TFT_MISO=19
-D TFT_MOSI=23
-D TFT_SCLK=18
-D TFT_CS=15
-D TFT_DC=2
-D TFT_RST=4
-D TFT_TOUCH_CS=21
-D LOAD_GLCD=1
-D LOAD_FONT2
-D LOAD_FONT4
-D LOAD_FONT6
-D LOAD_FONT7
-D LOAD_FONT8
-D LOAD_GFXFF
-D SMOOTH_FONT
-D SPI_FREQUENCY=27000000
</code>
and in the User_Setup.h I have defined
<code>
#define TOUCH_CS 21 // Chip select pin (T_CS) of touch screen
</code>
With the ILI9341_2 driver the display works perfectly, but
I'm doing something wrong or have overlooked something.
Thank you

**Comment by Bodmer** [comment]
-D TFT_TOUCH_CS=21

Should read:

-D  TOUCH_CS=21

**Comment by MHz000** [comment]
Thank you very much Bodmer for the quick answer, the error, my stupid mistake, is now fixed. The display shows what it is supposed to do. **chapeau!**

But now other error messages appear
<code>
'File' was not declared in this scope
'f' was not declared in this scope
</code>
even though I included that.
  <code>
#include <FS.h>
</code>

Another question concerns the 'redfined' warnings for the build_flags set in the platformio.ini:
<code>
build_flags =
-D USER_SETUP_LOADED
-D ILI9341_2_DRIVER
-D TFT_MISO=19
-D TFT_MOSI=23
-D TFT_SCLK=18
-D TFT_CS=15
-D TFT_DC=2
-D TFT_RST=4
-D TOUCH_CS=21
-D LOAD_GLCD=1
-D LOAD_FONT2
-D LOAD_FONT4
-D LOAD_FONT6
-D LOAD_FONT7
-D LOAD_FONT8
-D LOAD_GFXFF
-D SMOOTH_FONT
-D SPI_FREQUENCY=27000000
</code>
Hopefully not another stupid question.

Thank you for this library and the time you spent with user support

**Reply by Bodmer** [reply]
You have to assign values for the define to be recognised:
-D USER_SETUP_LOADED=1
-D ILI9341_2_DRIVER=1

And:
-D LOAD_FONT2=1
-D LOAD_FONT4=1
-D LOAD_FONT6=1
-D LOAD_FONT7=1
-D LOAD_FONT8=1
-D LOAD_GFXFF=1
-D SMOOTH_FONT=1

Which example sketch gives that error?

**Comment by MHz000** [comment]
Bodmer I use the example
<code>
https://github.com/Bodmer/TFT_eSPI/blob/master/examples/320%20x%20240/Keypad_240x3/Keypad_240x320.ino
</code>
my source code contains
<code>
#include <Arduino.h>
#include <TFT_eSPI.h>
#include <User_Setup.h>
#include <FS.h>
_//include is ok it is not marked as missing_
TFT_eSPI tft = TFT_eSPI();
</code>
and the platform.ini looks like this
<code>
Platform.ini
[env:esp32doit-devkit-v1]
platform = espressif32
board = esp32doit-devkit-v1
framework = arduino
monitor_speed = 115200
upload_speed = 512000

lib_deps = bodmer/TFT_eSPI@^2.5.31 <FS.h

build_flags =
-D USER_SETUP_LOADED=1
-D ILI9341_2_DRIVER=1
-D TFT_MISO=19
-D TFT_MOSI=23
-D TFT_SCLK=18
-D TFT_CS=15
-D TFT_DC=2
-D TFT_RST=4
-D TOUCH_CS=21
-D LOAD_GLCD=1
-D LOAD_FONT2=1
-D LOAD_FONT4=1
-D LOAD_FONT6=1
-D LOAD_FONT7=1
-D LOAD_FONT8=1
-D LOAD_GFXFF=1
-D SMOOTH_FONT=1
-D SPI_FREQUENCY=27000000
</code>

What surprises me is that #include <FS.h> is not noted and I cannot find and include such a file. 
You may see from this that I have little experience. 
Thank you for your patience and support.
